src/main/323C.c

```diff
@@ -1469,7 +1467,26 @@ INCLUDE_ASM("asm/us/main/nonmatchings/323C", func_8002B160);
 
 INCLUDE_ASM("asm/us/main/nonmatchings/323C", func_8002B1E8);
 
-INCLUDE_ASM("asm/us/main/nonmatchings/323C", func_8002B288);
+void func_8002B288(struct Unk* arg0)
+{
+    s32 temp_v0;
+    u16 var_v0;
+    s8 temp_v1;
+    u16 var_a0;
+
+    arg0->unk3 = 0;
+    if (arg0->unk14 < 0) {
+        var_v0 = arg0->x_pos;
+        var_a0 = arg0->y_pos;
+    } else {
+        temp_v0 = arg0->unk14 * 0x54 / 2;
+        var_v0 = arg0->x_pos - D_801419BA[temp_v0];
+        var_a0 = arg0->y_pos - D_801419BE[temp_v0];
+    }
+    if ((((var_v0 + 0x20) & 0xFFFF) < 0x180U) && (((var_a0 + 0x20) & 0xFFFF) < 0x130U)) {
```

Comment:
There's probably a cleaner way to write this but I haven't found it yet.

Reply:
Try this? https://decomp.me/scratch/zemvr

Reply:
Thanks!